internal/controller/ols_app_server_assets.go

```diff
@@ -299,6 +299,52 @@ func (r *OLSConfigReconciler) generateServiceMonitor(cr *olsv1alpha1.OLSConfig)
 	return &serviceMonitor, nil
 }
 
+func (r *OLSConfigReconciler) generatePrometheusRule(cr *olsv1alpha1.OLSConfig) (*monv1.PrometheusRule, error) {
+	metaLabels := generateAppServerSelectorLabels()
+	metaLabels["app.kubernetes.io/component"] = "metrics"
+
+	rule := monv1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      AppServerPrometheusRuleName,
+			Namespace: r.Options.Namespace,
+			Labels:    metaLabels,
+		},
+		Spec: monv1.PrometheusRuleSpec{
+			Groups: []monv1.RuleGroup{
+				{
+					Name: "ols.operations.rules",
+					Rules: []monv1.Rule{
+						{
+							Record: "ols:rest_api_query_calls_total:2xx",
+							Expr:   intstr.FromString("sum by(status_code) (ols_rest_api_calls_total{path=\"/v1/query\",status_code=~\"2..\"})"),
+							Labels: map[string]string{"status_code": "2xx"},
```

Comment:
this is prometheus relabeling logic, it converts all the 200/201/202/etc status codes, to 2xx, so that we have a single label value instead of many, which reduces cardinality.

Reply:
I see, we want a single cardinality from status code on this metric. No problem then :)